universum/configuration_support.py

```diff
@@ -116,30 +117,30 @@ class Variations(list):
 
     """
 
-    def __add__(self, other):
+    def __add__(self, other: List) -> 'Variations':
```

Comment:
typing 'other' as 'Variations' here generates mypy error:
universum/configuration_support.py:120: error: Argument 1 of "__add__" is incompatible with supertype "list"; supertype defines the argument type as "List[Any]"
universum/configuration_support.py:120: note: This violates the Liskov substitution principle
universum/configuration_support.py:120: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides

Reply:
yep. We override the method, but narrow down the expected argument, which means that Variations can't seamlessly replace instances of List, which violates Liskov subs. principle

Reply:
that's why initially I used List for __mul__ too, since I assumes it should work for any List